docs/reference/data-management/migrate-index-allocation-filters.asciidoc

```diff
@@ -0,0 +1,33 @@
+[role="xpack"]
+[[migrate-index-allocation-filters]]
+== Migrate index allocation filters to node roles
+
+If you currently use custom node attributes and routing allocation filters to move indices
+through data tiers in a hot-warm-cold architecture, we recommend that you switch to using node roles.
+The built-in node roles enable {ilm-init} to automatically migrate indices between data tiers.
+You no longer need to explicitly invoke the allocate action in your {ilm-init} policies.
+
+NOTE: While index allocation filters are no longer the best way to manage your data
+in a hot-warm-cold architecture, you can still use them to control shard allocation for other purposes.
+
+To switch to using node roles:
+
+. Assign your nodes to the appropriate data tier using the `data_hot`, `data_warm`,
+`data_cold`, and `data_frozen` roles.
+When you add a data tier to an {ess} deployment, this is done automatically.
+You can change the role of an existing node in an {ess} deployment with the
+{cloud}/ec-api-deployment-crud.html#ec_update_a_deployment[Update deployment API].
+If you are directly managing your own cluster,
+configure the appropriate data role for each node in `elasticsearch.yml`.
+. Update your {ilm-init} policy to:
+.. Remove the {ilm-init} allocate action from each phase in your index lifecycle policy.
+.. Ensure the policy specifies the corresponding phase for each data tier in your architecture.
+For example, if you enable the warm and cold data tiers for a deployment, your policy must
+include the hot, warm, and cold phases.
+This is necessary for {ilm-init} to automatically move indices through the data tiers,
+even if you don't need to perform any other actions and the phase is empty.
+. Update your index template to remove the custom node attribute you used to assign new indices
+to the hot tier. New indices are automatically assigned to `data_content` nodes, and new backing
+indices for data streams are automatically assigned to `data_hot` nodes.
+. To enable {ilm} to automatically move existing managed indices through the data tiers,
+remove your routing allocation filters from those indices.
```

Comment:
I think we need to be a bit more verbose here and link this scenario into the previously described one (where there's a mix of an index template that configures node attributes routing with the `7.10+` releases that default to node roles allocation - https://www.elastic.co/guide/en/elasticsearch/reference/current/data-tiers.html#data-tier-migration and https://www.elastic.co/guide/en/elasticsearch/reference/current/data-tiers.html#data-tier-allocation)